python/xir/program.py

```diff
@@ -258,167 +260,252 @@ class XIRProgram:
 
-        valid_match = re.match(r"^\d+\.\d+\.\d+$", version)
-        if valid_match is None or valid_match.string != version:
-            raise ValueError(
-                f"Invalid version number {version} input. Must be SemVer style (MAJOR.MINOR.PATCH)."
-            )
         self._version = version
         self._use_floats = use_floats
 
-        self._include = []
-        self._options = dict()
+        self._includes = []
+        self._options = {}
         self._statements = []
 
-        self._declarations = {"gate": [], "func": [], "output": [], "operator": []}
+        self._declarations = {key: [] for key in ("gate", "func", "output", "operator")}
 
-        self._gates = dict()
-        self._operators = dict()
-        self._variables = set()
+        self._gates = {}
+        self._operators = {}
 
-        self._called_ops = set()
+        self._variables = set()
+        self._called_functions = set()
 
     def __repr__(self) -> str:
+        """Returns a string representation of the XIR program."""
         return f"<XIRProgram: version={self._version}>"
 
     @property
     def version(self) -> str:
-        """Version number of the program
+        """Returns the version number of the XIR program.
 
         Returns:
-            str: program version number
+            str: version number
         """
         return self._version
 
     @property
     def use_floats(self) -> bool:
+        """Returns whether floats and complex types are used by the XIR program.
+
+        Returns:
+            bool: whether floats and complex types are used
+        """
         return self._use_floats
 
     @property
-    def wires(self) -> Set[int]:
-        """Get the wires of an XIR program"""
+    def wires(self) -> Collection[Wire]:
+        """Returns the wires of the XIR program.
+
+        Returns:
+            Collection[Wire]: collection of wires
+        """
         wires = []
         for stmt in self.statements:
             wires.extend(stmt.wires)
 
         return set(wires)
 
     @property
-    def include(self) -> List[str]:
-        """Included XIR libraries/files used in the program
+    def called_functions(self) -> Collection[str]:
+        """Returns the functions that are called in the XIR program.
 
         Returns:
-            list[str]: included libraries/files
+            Collection[str]: collection of function names
         """
-        return self._include
+        return self._called_functions
 
     @property
-    def options(self) -> Dict[str, Any]:
-        """Script-level options declared in the program
+    def declarations(self) -> Mapping[str, Sequence[Declaration]]:
+        """Returns the declarations in the XIR program.
 
         Returns:
-            Dict: declared scipt-level options
+            Mapping[str, Sequence[Declaration]]: dictionary of declarations
+                sorted into the following keys: 'gate', 'func', 'output' and
+                'operator'.
```

Comment:
I believe Sphinx doesn't like tabs in the "Returns:" docstring part. Not sure why exactly, but iirc PennyLane never uses tabs after linebreaks for the returns.
```suggestion
            Mapping[str, Sequence[Declaration]]: dictionary of declarations
            sorted into the following keys: 'gate', 'func', 'output' and
            'operator'.
```